pkg/datapath/loader/base.go

```diff
@@ -293,6 +332,41 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner,
 	log.Infof("Setting up base BPF datapath (BPF %s instruction set, %s clock source)",
 		args[initBPFCPU], clockSource[option.Config.ClockSource])
 
+	// EKS expects the aws daemonset to set up the following rules that
+	// relate to NodePort traffic between nodes:
+	//
+	// # sysctl -w net.ipv4.conf.eth0.rp_filter=2
+	// # iptables -t mangle -A PREROUTING -i eth0 -m comment --comment "AWS, primary ENI" -m addrtype --dst-type LOCAL --limit-iface-in -j CONNMARK --set-xmark 0x80/0x80
+	// # iptables -t mangle -A PREROUTING -i eni+ -m comment --comment "AWS, primary ENI" -j CONNMARK --restore-mark --nfmask 0x80 --ctmask 0x80
+	// # ip rule add fwmark 0x80/0x80 lookup main
+	//
+	// It marks packets coming from another node through eth0, and restores
+	// the mark on the return path to force a lookup into the main routing
+	// table. Without these rules, the "ip rules" set by the cilium-cni
+	// plugin tell the host to lookup into the table related to the VPC for
+	// which the CIDR used by the endpoint has been configured.
+	//
+	// We want to reproduce equivalent rules to ensure correct routing.
+	// See also iptables.addCiliumENIRules().
+	if option.Config.Masquerade && option.Config.IPAM == ipamOption.IPAMENI {
+		ifaces, err := getEgressMasqueradeInterfaces()
+		if err != nil {
+			return fmt.Errorf("failed to set rp_filter for interfaces with egress masquerading for ENI multi-node NodePort: %w", err)
+		}
+		for _, iface := range ifaces {
+			sysSettings = append(sysSettings,
+				setting{"net.ipv4.conf." + iface + ".rp_filter", "2", false})
+		}
+		if err := route.ReplaceRule(route.Rule{
+			Priority: linux_defaults.RulePriorityEgress - 1,
```

Comment:
Regarding point (3), I think it would be valuable to create this as a const. It's better to codify a value that we depend on, so that it is discoverable when debugging, understanding docs, etc.

Reply:
(Seeing the marks and masks down below are also not consts, but maybe that's something we've always done with marks and masks? Either way, I think defining the rule priority still makes sense as there's precedent.)

Reply:
Yes, collecting these in canonical locations with all relevant references (+ comments on how they were chosen) would be useful for maintenance.